akka-stream/src/main/scala/akka/stream/scaladsl/Graph.scala

```diff
@@ -571,6 +571,103 @@ final class MergeSorted[T: Ordering] extends GraphStage[FanInShape2[T, T, T]] {
   }
 }
 
+/**
+ * Merge multiple pre-sorted streams such that the resulting stream is sorted.
+ *
+ * The stage pulls and buffers one element from each of the given inlets. It does not emit until
+ * all inlets have an available element (or have been closed) at which point it will emit the smallest
+ * element and pull the corresponding inlet.
+ *
+ * '''Emits when''' all inputs have an element available
+ *
+ * '''Backpressures when''' downstream backpressures
+ *
+ * '''Completes when''' all upstreams complete
+ *
+ * '''Cancels when''' downstream cancels
+ */
+object MergeSortedN {
+
+  /**
+   * Create a new `MergeSortedN`.
+   */
+  def apply[A: Ordering](n: Int) = new MergeSortedN[A](n)
+}
+
+/**
+ * INTERNAL API
+ */
+@InternalApi private[akka] final class MergeSortedN[T: Ordering](inputPorts: Int)
```

Comment:
Add eagerComplete?

Reply:
I don't want to spend more time on this PR than making it mergeable so that can be a follow up.